File: 0303-range-sum-query-immutable/0303-range-sum-query-immutable.java
class NumArray {

    private int[] sums;
    public NumArray(int[] nums) {
        int N = nums.length;
        sums = new int[nums.length];
        int index = 0;
        for(int number : nums) sums[index++] += number;
        
    }
    
    public int sumRange(int left, int right) {
        int total = 0;
        while(left <= right)
        {
            total += sums[left++];
        }
        return total;
    }
}

/**
 * Your NumArray object will be instantiated and called as such:
 * NumArray obj = new NumArray(nums);
 * int param_1 = obj.sumRange(left,right);
 */

File: 0303-range-sum-query-immutable/0303-range-sum-query-immutable.py
class NumArray:

    def __init__(self, nums: List[int]):
        self.dp = [0] * (len(nums) + 1)
        i = 0
        self.dp[i] = 0

        for num in nums:
            i += 1
            self.dp[i] = self.dp[i-1] + num

    def sumRange(self, left: int, right: int) -> int:
        return self.dp[right + 1] - self.dp[left]
        
        


# Your NumArray object will be instantiated and called as such:
# obj = NumArray(nums)
# param_1 = obj.sumRange(left,right)